File: Java/Week_1/Day7Loops.java
public class Day7Loops {
    public static void main(String[] args) {
        
        // for loop - repeats a fixed number of times
        System.out.println("--- For Loop ---");
        for (int i = 0; i < 5; i++) {
            System.out.println("Count: " + i);
        }
        
        // while loop
        System.out.println("--- While Loop ---");
        int count = 0;
        while (count < 5) {
            System.out.println("While count: " + count);
            count++;   // must update, or infinite loop!
        }
        
        // loop through an array
        System.out.println("--- Loop through array ---");
        String[] fruits = {"apple", "banana", "mango"};
        for (String fruit : fruits) {
            System.out.println(fruit);
        }
        
        // break and continue
        System.out.println("--- Break and Continue ---");
        for (int i = 0; i < 10; i++) {
            if (i == 5) {
                break;
            }
            System.out.println("Break demo: " + i);
        }
        
        for (int i = 0; i < 5; i++) {
            if (i == 2) {
                continue;
            }
            System.out.println("Continue demo: " + i);
        }
    }
}

File: Python/Week_1/day7_loops.py
# Day 7 - Loops in Python

# for loop - repeats a fixed number of times
print("--- For Loop ---")
for i in range(5):
    print("Count:", i)   # prints 0,1,2,3,4

# for loop with a range start/stop/step
for i in range(1, 10, 2):
    print("Odd number:", i)   # 1,3,5,7,9

# while loop - repeats while a condition is true
print("--- While Loop ---")
count = 0
while count < 5:
    print("While count:", count)
    count += 1   # IMPORTANT: must update, or infinite loop!

# loop through a list
print("--- Loop through list ---")
fruits = ["apple", "banana", "mango"]
for fruit in fruits:
    print(fruit)

# break and continue
print("--- Break and Continue ---")
for i in range(10):
    if i == 5:
        break        # stops the loop completely
    print("Break demo:", i)

for i in range(5):
    if i == 2:
        continue     # skips just this iteration, loop continues
    print("Continue demo:", i)